### ISSUE TITLE
Weight function in similarity calculation

### ISSUE THREAD

**Issue by qzhu2017** [qzhu2017/XRD#13]
@sayred1 
Right now, the triangle function is used. 
<img width="475" alt="image" src="https://user-images.githubusercontent.com/29445366/72691724-6af6d980-3adc-11ea-9256-636f1a3b1fd2.png">

**l in the above equation should be consistent to r_range values. Why did you set 0.6 here?**.
https://github.com/qzhu2017/XRD/blob/ca73701349c440f171eb290c52553613d1f40cc3/similarity.py#L10 

However, triangle function is not smooth. I suggest you also add the cosine function used in our PyXtal_FF
<img width="423" alt="image" src="https://user-images.githubusercontent.com/29445366/72691658-f2901880-3adb-11ea-8489-07376197b5e9.png">

This can ensure that the function smoothly goes to zero when r is approaching to r_cut.

**Comment by sayred1** [comment]
@qzhu2017 I chose the value of 0.6 based off of the [de Gelder paper](https://onlinelibrary.wiley.com/doi/full/10.1002/1096-987X%28200102%2922%3A3%3C273%3A%3AAID-JCC1001%3E3.0.CO%3B2-0).

"The parameter l defines the width (degrees 2θ) of the neighborhood taken into account. This function extracts information from the central part of the crosscorrelation function with a weight that decreases proportionally to the distance from the origin (r=0). After a number of experiments it was found that values of l between 0.6 and 3.0 lead to stable and comparable results for the powder diffractions patterns of the 20 complexes."

However since our defined range of r may differ from theirs (which they do not mention), we should test our own values of l/R_cut?

**Comment by qzhu2017** [comment]
@sayred1 
Yes, but I meant to say that if you set +/-1 for `r_range`, then `l` should be 1 (not 0.6). Don't mess up the relation. 
In this case, I suggest we don't use dictionary to denote `weight`, just use string.

```python
def __init__(self, f, g, N = None, x_range = None, r_range = None, weight = 'triangle'):
```

**Comment by qzhu2017** [comment]
@sayred1  I just fixed it in commit 69758650c6b65877233c10ae05cdcdca1f5c553b
